apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletGlobalState.java

```diff
@@ -33,7 +35,7 @@
 @GlobalState
 public class ServletGlobalState {
 
-    public static final Set<String> nameInitialized = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
+    public static final WeakConcurrentMap<ClassLoader, Boolean> nameInitialized = WeakMapSupplier.createMap();
```

Comment:
Just FYT, in #1303, I've create a `WeakMapSupplier.createSet()` method.

Reply:
If I remember that after merging #1303, I will switch to using it instead.